### ISSUE TITLE
Getting timeout on establishing a tunnel

### ISSUE THREAD

**Issue by iramello** [mludvig/aws-ssm-tools#27]
Hi,
Do you know why running something like this in Ubuntu 20.04 (python 3.8) always fails with timeout?
Thanks

```
ivan@anecua:~$ ssm-tunnel i-7171717171717171 --profile my-profile --debug
[ssm-tunnel] DEBUG: Logging level set to DEBUG
[ssm-tunnel] DEBUG: Spawning: aws --profile my-profile  ssm start-session --target i-7171717171717171
[ssm-tunnel] DEBUG: PID: 288300
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/pexpect/expect.py", line 111, in expect_loop
    incoming = spawn.read_nonblocking(spawn.maxread, timeout)
  File "/usr/lib/python3/dist-packages/pexpect/pty_spawn.py", line 482, in read_nonblocking
    raise TIMEOUT('Timeout exceeded.')
pexpect.exceptions.TIMEOUT: Timeout exceeded.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/ssm-tunnel", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/ssm_tools/ssm_tunnel_cli.py", line 345, in main
    tunnel = SsmTunnel(instance_id, profile=args.profile, region=args.region, logger_name=logger_name)
  File "/usr/local/lib/python3.8/dist-packages/ssm_tools/ssm_tunnel_cli.py", line 83, in __init__
    super().__init__(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/ssm_tools/talker.py", line 9, in __init__
    self.connect(instance_id, profile, region)
  File "/usr/local/lib/python3.8/dist-packages/ssm_tools/talker.py", line 23, in connect
    self.wait_for_prompt()
  File "/usr/local/lib/python3.8/dist-packages/ssm_tools/talker.py", line 49, in wait_for_prompt
    self._child.expect('sh.*\$ $')
  File "/usr/lib/python3/dist-packages/pexpect/spawnbase.py", line 340, in expect
    return self.expect_list(compiled_pattern_list,
  File "/usr/lib/python3/dist-packages/pexpect/spawnbase.py", line 369, in expect_list
    return exp.expect_loop(timeout)
  File "/usr/lib/python3/dist-packages/pexpect/expect.py", line 119, in expect_loop
    return self.timeout(e)
  File "/usr/lib/python3/dist-packages/pexpect/expect.py", line 82, in timeout
    raise TIMEOUT(msg)
pexpect.exceptions.TIMEOUT: Timeout exceeded.
<pexpect.pty_spawn.spawn object at 0x7f81cf756100>
command: /usr/local/bin/aws
args: [b'/usr/local/bin/aws', b'--profile', b'my-profile', b'ssm', b'start-session', b'--target', b'i-7171717171717171']
buffer (last 100 chars): '\r\nStarting session with SessionId: my-user@company-097fbec8ab0500248\r\n$ '
before (last 100 chars): '\r\nStarting session with SessionId: my-user@company-097fbec8ab0500248\r\n$ '
after: <class 'pexpect.exceptions.TIMEOUT'>
match: None
match_index: None
exitstatus: None
flag_eof: False
pid: 288300
child_fd: 5
closed: False
timeout: 10
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('sh.*\\$ $')
```

**Comment by mludvig** [comment]
What's the OS on instance i-7171717171717171? Looks like the shell prompt is only `$ ` instead of e.g. `sh-4.2 $ ` as on Amazon Linux 2. That's why ssm-tunnel keeps waiting for the shell prompt, it doesn't recognize the prompt that your instance is showing. What's the OS? I'll try to replicate it myself.

**Comment by fdx-apdewis** [comment]
I also see this problem with ubuntu based instances (20.04 specifically), which as above only has $ for the prompt.
Is it possible to regex match on this to make any leading characters prior to the $ optional?

**Comment by fdx-apdewis** [comment]
A work around is possible by following the instructions here https://docs.aws.amazon.com/systems-manager/latest/userguide/session-preferences-shell-config.html. 

and putting  

PS1='sh4.2$ '

In the Linux shell profile section, which is necessary to force change the prompt as SSM sessions ignore the local .profile config files.

**Comment by mludvig** [comment]
It's fixed now and will be released in 1.5.1 very soon. Thanks for your patience.
